src/services/mentors.js

```diff
@@ -906,10 +888,21 @@ module.exports = class MentorsHelper {
 						cachedProfile.image = await utils.getDownloadableUrl(cachedProfile.image)
 					} catch (error) {
 						console.error(`Failed to get downloadable URL for cached profile image:`, error)
-						cachedProfile.image = null
 					}
 				}
 
+				let communications = null
+				if (cachedProfile?.meta?.communications_user_id) {
+					try {
+						const chat = await communicationHelper.login(id, tenantCode)
+						communications = chat
+					} catch (error) {}
+				}
+				cachedProfile.meta = {
+					...cachedProfile.meta,
+					communications,
+				}
+
```

Comment:
This looks like an unnecessary reassignment and could end up unintentionally overwriting meta.communications when a login fails.

Also, wouldn’t it make more sense to write it to the cache when it’s not found?